clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/util/id/DcosSpinnakerAppId.groovy

```diff
@@ -0,0 +1,153 @@
+package com.netflix.spinnaker.clouddriver.dcos.deploy.util.id
+
+import static com.google.common.base.Preconditions.checkArgument
+import static com.google.common.base.Strings.nullToEmpty
+
+import com.netflix.frigga.Names
+
+/**
+ * Represents a hierarchical Spinnaker specific application identifier for DCOS.
+ * Structure - /account/region/app-stack-detail-sequence
+ */
+class DcosSpinnakerAppId {
+    public final static def SAFE_REGION_SEPARATOR = "_"
+
+    private final def marathonPath
+
+    public DcosSpinnakerAppId(final String marathonAppId) {
```

Comment:
I still think the static factory methods are much more descriptive in this case.

We're transforming from a string to a DcosSpinnakerAppId, which parse describes (many classes do this - Integer.parseInt, Boolean.parseBoolean, etc.). It also carries the connotation just by looking at the name that it can possibly fail. From just the constructor, you don't get that information.

From there, I don't think it's a stretch to change the (account, region, name) constructor to a static factory method for consistency.